### PR TITLE
Fix IPEX auto importer

### PR DESCRIPTION
**Pull request by qiyuangong**
## Description

* Fix ipex auto importer with builtins.
* Raise importError if the user imports ipex manually before importing ipex_llm. Do nothing if they import ipex after importing ipex_llm.
* Remove import ipex in examples. 

### 4. How to test?
- [ ] N/A
- [ ] Unit test
- [ ] Application test
- [ ] Document test
- [ ] ...

### 5. New dependencies

- [ ] New Python dependencies
       - Dependency1 
       - Dependency2
       - ...
- [ ] New Java/Scala dependencies and their license
       - Dependency1 and license1
       - Dependency2 and license2
       - ...

### CODE REVIEW

File: python/llm/src/ipex_llm/utils/ipex_importer.py [hunk 1]
```diff
@@ -56,12 +56,22 @@ def import_ipex(self):
...
         if self.is_xpu_version_installed():
+            ipex_imported = False
+            try:
+                distribution('intel_extension_for_pytorch')
+                ipex_importer = True
+            except PackageNotFoundError:
+                pass
+            if ipex_imported:
+                logging.warning("IPEX-LLM will automatically import Intel Extension for PyTorch.",
+                                "Please avoid import Intel Extension for PyTorch manually!")
```
Comment (jason-dai) [inline python/llm/src/ipex_llm/utils/ipex_importer.py:67]: Report error "Please import ipex-llm before import Intel Extension for PyTorch"

File: python/llm/src/ipex_llm/utils/ipex_importer.py [hunk 2]
```diff
@@ -74,8 +84,12 @@ def get_ipex_version(self):
         # try to import Intel Extension for PyTorch and get version
         try:
             import intel_extension_for_pytorch as ipex
+            if ipex is not None:
+                # Expose ipex to Python builtins
+                builtins.ipex = ipex
             self.ipex_version = ipex.__version__
         except ImportError:
+            logging.warning("Can not import intel_extension_for_pytorch.")
```
Comment (jason-dai) [inline python/llm/src/ipex_llm/utils/ipex_importer.py:92]: shall we report error here?

Comment (qiyuangong) [reply python/llm/src/ipex_llm/utils/ipex_importer.py:88]: Yes. Change to raise errors.

File: python/llm/src/ipex_llm/utils/ipex_importer.py (hunk 1 above)
Comment (qiyuangong) [reply python/llm/src/ipex_llm/utils/ipex_importer.py]: Done

File: python/llm/src/ipex_llm/utils/ipex_importer.py [hunk 3]
```diff
@@ -51,15 +54,36 @@ def is_xpu_version_installed():
...
-        Raises ImportError if failed
+        Raises ImportError and invalidInputError if failed
         """
         if self.is_xpu_version_installed():
-            import intel_extension_for_pytorch as ipex
+            # Check if user import ipex manually
+            if 'ipex' in sys.modules or 'intel_extension_for_pytorch' in sys.modules:
+                logging.error("IPEX-LLM will automatically import Intel Extension for PyTorch.")
+                log4Error.invalidInputError(False,
+                                            "Please import ipex-llm before importing \
```
Comment (jason-dai) [inline python/llm/src/ipex_llm/utils/ipex_importer.py:66]: ipex_llm

File: python/llm/src/ipex_llm/utils/ipex_importer.py [hunk 4]
```diff
@@ -51,15 +54,36 @@ def is_xpu_version_installed():
...
+        Try to import Intel Extension for PyTorch as ipex for XPU
 
-        Raises ImportError if failed
+        Raises ImportError and invalidInputError if failed
         """
         if self.is_xpu_version_installed():
-            import intel_extension_for_pytorch as ipex
+            # Check if user import ipex manually
+            if 'ipex' in sys.modules or 'intel_extension_for_pytorch' in sys.modules:
+                logging.error("IPEX-LLM will automatically import Intel Extension for PyTorch.")
```
Comment (jason-dai) [inline python/llm/src/ipex_llm/utils/ipex_importer.py:64]: Please import ipex_llm before importing intel_extension_for_pytorch

Comment (qiyuangong) [reply python/llm/src/ipex_llm/utils/ipex_importer.py:64]: Done

File: python/llm/src/ipex_llm/utils/ipex_importer.py (hunk 3 above)
Comment (qiyuangong) [reply python/llm/src/ipex_llm/utils/ipex_importer.py]: Addressed

**Review by Oscilloscope98 — APPROVED** [review]

File: python/llm/src/ipex_llm/utils/ipex_importer.py [hunk 5]
```diff
@@ -51,15 +54,36 @@ def is_xpu_version_installed():
...
+        Try to import Intel Extension for PyTorch as ipex for XPU
 
-        Raises ImportError if failed
+        Raises ImportError and invalidInputError if failed
         """
         if self.is_xpu_version_installed():
-            import intel_extension_for_pytorch as ipex
+            # Check if user import ipex manually
+            if 'ipex' in sys.modules or 'intel_extension_for_pytorch' in sys.modules:
+                logging.error("ipex_llm will automatically import intel_extension_for_pytorch.")
```
Comment (jason-dai) [inline python/llm/src/ipex_llm/utils/ipex_importer.py:64]: Please import ipex_llm before importing intel_extension_for_pytorch

**Pull request merged**